public/app/features/transformers/prepareTimeSeries/prepareTimeSeries.ts

```diff
@@ -282,6 +284,20 @@ export function toTimeSeriesLong(data: DataFrame[]): DataFrame[] {
   return result;
 }
 
+export function longToMultiTimeSeries(frame: DataFrame): DataFrame[] {
+  // All the string fields
+  const matcher = (field: Field) => field.type === FieldType.string;
```

Comment:
^^ replacing https://github.com/grafana/grafana/pull/64474/files#diff-7b2888e194184dd19ca6cac1fea261e2030ab76c1b8fa5bb471b1bca343894bfL206

This lets us set the matcher explicitly rather than with a double config lookup

Reply:
Nice, this looks better, I see how you use that matcher with the transformation now! Cool 😎